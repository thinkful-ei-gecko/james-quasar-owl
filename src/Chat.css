.chat-container {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.chat-feed {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.chat-feed > li {
  display: flex;
  line-height: 2em;
  color: grey;
}

p {
  padding: 2px;
}

input {
  width: 92%;
  margin: 0 auto;
  padding: 12px;
}

.chat-avatar {
  height: 50px;
  width: 50px;
  border-radius: 100%;
}

.chat-name {
  font-weight: bolder;
  margin-right: 4px;
}