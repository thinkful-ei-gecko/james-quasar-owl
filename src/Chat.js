import React from 'react';
import store from './store.js';

export default function Chat(props) {

  const toDate = function(time) {
    const formattedTime = new Date(time);
    return formattedTime.toLocaleString('en-US', { hour: 'numeric', minute: 'numeric', hour12: true });
  }

  const getParticipantName = function(event) {
    const id = event.participantId; 
    return store.participants[id - 1].name;
  };

  const getParticipantAvatar = function(event) {
    const id = event.participantId; 
    return store.participants[id - 1].avatar;
  };

  const createEvent = function(event) {

    let message = ''; 
    switch (event.type) {
      case 'thumbs-up':
        message = 'gave a thumbs up';
        break;
      case 'thumbs-down':
        message = 'gave a thumbs down';
        break;
      case 'clap':
        message = 'clapped';
        break;
      case 'join':
        message = 'joined';
        break;
      case 'leave':
        message = 'left';
        break;
      case 'join-stage':
        message = 'joined stage';
        break;
      case 'leave-stage':
        message = 'left stage';
        break;
      default:
        message = 'unknown';
        break;
    }
    return (
      <li key={event.timestamp}><span className="chat-name">{getParticipantName(event)}</span> {message}</li>
    );
  };

  const createMessage = function(event) {
    return (
      <li key={event.timestamp}>
        <img src={getParticipantAvatar(event)} alt={getParticipantName(event) + ' avatar'} />
        <p><span className="chat-name">{getParticipantName(event)}</span> {toDate(event.time)}</p>
        <p>{event.message}</p>
      </li>
    );
  };

  return (
    <ul>
    {
      store.chatEvents.map(event => (
          event.type === 'message' ? createMessage(event) : createEvent(event)
      ))
    }
    </ul>

  );
}