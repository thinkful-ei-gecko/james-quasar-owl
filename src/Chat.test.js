import React from 'react';
import ReactDOM from 'react-dom';
import Chat from './Chat';
import renderer from 'react-test-renderer';

describe('Chat tests', () => {
  it('renders without breaking', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Chat key={1}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render same UI', () => {
    const tree = renderer.create(<Chat key={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

})