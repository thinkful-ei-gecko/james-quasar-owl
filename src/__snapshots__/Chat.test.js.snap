// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat tests render same UI 1`] = `
<ul>
  <li>
    <img
      alt="Axe Kubicka avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    <p>
      <span
        className="chat-name"
      >
        Axe Kubicka
      </span>
       
      4:50 AM
    </p>
    <p>
      Hello world
    </p>
  </li>
  <li>
    <span
      className="chat-name"
    >
      Caty Flucker
    </span>
     
    gave a thumbs up
  </li>
  <li>
    <span
      className="chat-name"
    >
      Frank Runciman
    </span>
     
    gave a thumbs down
  </li>
  <li>
    <span
      className="chat-name"
    >
      Koren Templeton
    </span>
     
    unknown
  </li>
  <li>
    <span
      className="chat-name"
    >
      Koren Templeton
    </span>
     
    clapped
  </li>
  <li>
    <span
      className="chat-name"
    >
      Koren Templeton
    </span>
     
    joined
  </li>
  <li>
    <span
      className="chat-name"
    >
      Ashla Attwool
    </span>
     
    left
  </li>
  <li>
    <span
      className="chat-name"
    >
      Axe Kubicka
    </span>
     
    joined stage
  </li>
  <li>
    <span
      className="chat-name"
    >
      Axe Kubicka
    </span>
     
    left stage
  </li>
</ul>
`;
